zap/src/main/java/org/zaproxy/zap/extension/search/SearchThread.java

```diff
@@ -33,6 +33,8 @@
 import org.parosproxy.paros.model.Session;
 import org.parosproxy.paros.network.HttpMalformedHeaderException;
 import org.parosproxy.paros.network.HttpMessage;
+import org.zaproxy.zap.extension.pscan.PassiveScanParam;
+import org.zaproxy.zap.extension.pscan.scanner.RegexAutoTagScanner;
```

Comment:
This should be pluggable, if/when the passive scanner is moved to the add-on it should not require further changes other than call the methods from the add-on.

Reply:
I'm not sure how I'd do that because we need to be able to get the details from the "rule". Unless I close this for now and re-implement after the move so that it only uses add-on classes (which is totally fine with me).

It could be done via a custom search like the fuzzer I guess, but I haven't dug into that.

Reply:
The search will need changes either way, since tags are part of the message I don't think the passive scanner is a requirement to search for tags (might be for the highlight but that's secondary). You just need to allow to set something that tells the highlight from/for a specific tag.

Reply:
If we simply don't highlight (which is totally fine from my point of view), then there wouldn't be a need to rely on the 'rules'. Shall I do away with highlighting for now?

Reply:
Sounds good to me.